Kubernetes deployments for version 7.1.0

icanci01 opened the pull request.

icanci01 committed:
Update README and add a way to build all the images and publish them to a user specified registry

icanci01 committed:
Add namespace and mysql database creation

icanci01 committed:
Initial services manifest files

icanci01 committed:
Add kubernetes dashboard for easier management
# Do not use in production #
Updated README with more info

icanci01 committed:
Update mysql deployment file
- Add configmaps
- Add init scripts
- Add readiness and liveness probes
- Small update README
- Add deploy/clear deployments .bat files

icanci01 committed:
Job configuration-utility adjusted to run for kubernetes
- Added the application.yml properites inside the job definition
- Adjusted the Dockerfile so that the propeties are alredy present in the image
- Adjusted the docker compose

icanci01 committed:
Update changes openncp-server + epsos-configuration folder
- Added a persistent volume for the epsos-configuration folder
- Added a job to copy the contents
- Modified the openncp-server deployment
- Made changes regarding the certificates

icanci01 committed:
Update all the deployments files
- eHealth Portal Backend
- eHealth Portal Frontend
- Gateway Backend
- Gateway Frontend
- TSC-STS
- OpenATNA

icanci01 committed:
Small updates regarding versions

psavva commented on build-artifacts/Dockerfile.openncp:
Please change this back

psavva commented on ehealth-portal-backend/manifests/ehealth-portal-backend-deployment.yaml:
```diff
@@ -0,0 +1,94 @@
+    spec:
+      containers:
+      - name: ehealth-portal-backend
+        image: docker.io/3ahealth/ehealth-portal-backend:7.1.0
```
Please remove 3aHealth registry here as this is a private registry and inaccessible for everyone.
Please use github container registry
https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry

icanci01 replied:
Done!

psavva commented on ehealth-portal-backend/manifests/ehealth-portal-backend-deployment.yaml:
```diff
@@ -0,0 +1,94 @@
+      containers:
+      - name: ehealth-portal-backend
+        image: docker.io/3ahealth/ehealth-portal-backend:7.1.0
+        ports:
```
This should not be needed, please review and verify

icanci01 replied:
Yes, you are right! Fixed!

psavva commented on ehealth-portal-backend/manifests/ehealth-portal-backend-deployment.yaml:
```diff
@@ -0,0 +1,94 @@
+            cpu: "1"
+        env:
+        - name: EPSOS_PROPS_PATH
+          value: "/opt/epsos-configuration/"
```
Please add this to the readme, the volume mount should be on every node that the services will run.

icanci01 replied:
Will adjust in a future PR where I will add ALL the details needed in the README! So far I added a lot of information but is not quite the final README.

psavva commented on openncp-client/manifests/openncp-client-deployment.yaml:
```diff
@@ -0,0 +1,94 @@
+            secretKeyRef:
+              name: openncp-secret
+              key: TLS_TRUSTSTORE_PWD
+        - name: JAVA_OPTS
```
This could be moved into a configmap for future improvements

icanci01 replied:
Will adjust in a future PR!

psavva commented on openncp-client/manifests/openncp-client-service.yaml:
```diff
@@ -0,0 +1,18 @@
+  - name: http
+    port: 7080
+    targetPort: 7080
+    nodePort: 30002
```
we do not tyically define nodePorts.  Typically the default is a LoadBalancer service type, and it's up to each admin installing the system to chose how they will install.  for now, this is OK

icanci01 replied:
Will adjust this in a future PR!

psavva commented on openncp-client/manifests/openncp-client-service.yaml:
```diff
@@ -0,0 +1,18 @@
+  - name: http
+    port: 7080
+    targetPort: 7080
+    nodePort: 30002
```
Typically we do not hard code the nodePort as it may be already assigned to another service.  Kubernetes manages the nodePort number dynamically, and thus doesn't need to be assigned.

icanci01 replied:
Commented the nodePort on each service! It is up to admins to define ports. Added also a comment for this.

psavva commented on openncp-configuration-utility/Dockerfile:
```diff
@@ -2,4 +2,7 @@ FROM eclipse-temurin:11-jre-jammy
 WORKDIR /usr/local/openncp-configuration-utility
 
+COPY openncp-configuration-utility.jar /usr/local/openncp-configuration-utility/openncp-configuration-utility.jar
+COPY openncp-configuration.properties /usr/local/openncp-configuration-utility/openncp-configuration.properties
```
This should be created as a configmap, and created as a Projected Volume

https://kubernetes.io/docs/concepts/storage/projected-volumes/

icanci01 replied:
Fixed!

psavva commented on openncp-gateway-backend/manifests/openncp-gateway-backend-deployment.yaml:
```diff
@@ -0,0 +1,94 @@
+            secretKeyRef:
+              name: mysql-secret
+              key: MYSQL_PASSWORD
+        - name: TLS_KEYSTORE_PATH
```
Please add instructions in the README how to generate the jks files

icanci01 replied:
Working on it! This will be added in a next PR where I will define more information in the readme!

icanci01 committed:
Update registry from 3aHealth to ghcr.io/ucy-ehealthlab
- Make usage of a publicly available container registry under the ucy-ehealthlab organization
- Update Registry build script with Portal Version
- Small change to README
- Update dashboard browser url

icanci01 committed:
Comment all nodeports (can be used for debugging before official deployment)
- Kubernetes manages the exposed ports. By using hardcoded ports, there might be port conflicts with other services running in the cluster.
- Will adjust to NodeBalancers in the future

icanci01 committed:
Remove assigned ports to containers! This seems to not be needed.

icanci01 committed:
Adjust resource allocations (requests and limits)

icanci01 committed:
Update configuration utility to make use of Projected Volume
